### PR TITLE
Add falling and rising hexagons in doom

### PR DESCRIPTION
**Pull request by peterhgombos**
DOOOOM!

### CODE REVIEW

File: src/DesertLayer.js [hunk 1]
```diff
@@ -576,7 +645,7 @@ function DandelionSeed(layer, scale) {
 
   dandelionSeed.wingCylinders = [];
 
-  Math.seedrandom("iverjo-is-sexy");
+  Math.seedrandom("Penis");
```
Comment (aleksanb) [inline src/DesertLayer.js]: :o:

Comment (peterhgombos) [inline src/DesertLayer.js]: Blame @Raane.

Comment (sigvef) [inline src/DesertLayer.js]: Don't change this unless you have a really good reason - you are risking changing the entire demo, if something is very random-dependent.

**Comment by peterhgombos** [comment]
fixed

File: src/DesertLayer.js [hunk 2]
```diff
@@ -439,6 +440,46 @@ DesertLayer.prototype.update = function(frame, relativeFrame) {
...
+  this.hexagonFallingTimings[3] = this.hexagonFallingBase + 60;
+  this.hexagonFallingTimings[8] = this.hexagonFallingBase + 70;
+
+
+  this.hexagonFallingTimings[11] = this.hexagonFallingBase + 30;
+  this.hexagonFallingTimings[21] = this.hexagonFallingBase + 40;
+  this.hexagonFallingTimings[31] = this.hexagonFallingBase + 50;
+  this.hexagonFallingTimings[40] = this.hexagonFallingBase + 60;
+  this.hexagonFallingTimings[50] = this.hexagonFallingBase + 70;
+  this.hexagonFallingTimings[60] = this.hexagonFallingBase + 80;
```
Comment (stianjensen) [inline src/DesertLayer.js]: Are all these calculations static, and possible to move out of the update loop?

**Comment by stianjensen** [comment]
In the storyboard, a crack in the ground goes from the edge toward the center. This would be really nice, so we can transition from the ground-falling effect to the water-lava effect.

File: src/DesertLayer.js [hunk 3]
```diff
@@ -439,6 +440,46 @@ DesertLayer.prototype.update = function(frame, relativeFrame) {
...
+  this.updateDoomHexagon(relativeFrame, this.hexagonFallingTimings[11], 11, {direction: 'fall'});
+  this.updateDoomHexagon(relativeFrame, this.hexagonFallingTimings[21], 21, {direction: 'fall'});
+  this.updateDoomHexagon(relativeFrame, this.hexagonFallingTimings[31], 31, {direction: 'fall'});
+  this.updateDoomHexagon(relativeFrame, this.hexagonFallingTimings[40], 40, {direction: 'fall'});
+  this.updateDoomHexagon(relativeFrame, this.hexagonFallingTimings[50], 50, {direction: 'fall'});
+  this.updateDoomHexagon(relativeFrame, this.hexagonFallingTimings[60], 60, {direction: 'fall'});
+
+  this.updateDoomHexagon(relativeFrame, this.hexagonFallingTimings[17], 17, {direction: 'up', speed: 30, maxHeight: 4000});
+
+  // this.desertHexes[11].material.color = new THREE.Color(0xff69b4)
```
Comment (cristeahub) [inline src/DesertLayer.js]: this line can be removed

File: src/DesertLayer.js [hunk 4]
```diff
@@ -521,6 +562,21 @@ DesertLayer.prototype.updateGrass = function(frame, relativeFrame) {
...
+  var t = clamp(0, relativeFrame - startFrame, 999999);
+  if (options.direction === 'up') {
+    hex.position.y = this.desertHexStart + options.speed * t < this.desertHexStart + options.maxHeight ? 
+                                                               this.desertHexStart + options.speed * t :  
+                                                               this.desertHexStart + options.maxHeight;
+  } else if (options.direction === 'fall') {
+    var acceleration = -5;
+    hex.position.y = this. desertHexStart + 0.5 * acceleration * t * t;
+
+    // hex.rotation.x = -Math.PI/2 + t / 60;
```
Comment (cristeahub) [inline src/DesertLayer.js]: this can be removed

File: src/DesertLayer.js [hunk 5]
```diff
@@ -576,7 +645,7 @@ function DandelionSeed(layer, scale) {
 
   dandelionSeed.wingCylinders = [];
 
-  Math.seedrandom("iverjo-is-sexy");
+  Math.seedrandom("sigveseb-is-dumb");
```
Comment (cristeahub) [inline src/DesertLayer.js]: maybe change this ^^

Comment (iver56) [inline src/DesertLayer.js]: O.o

**Comment by peterhgombos** [comment]
Updated

**Comment by peterhgombos** [comment]
And again, since the time offset of the start was too big.

File: src/DesertLayer.js (hunk 5 above)
Comment (sigvef) [inline src/DesertLayer.js]: -.-

**Pull request merged**